src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1934,9 +1934,36 @@ private class OffsetAndRule {
 
 
 #if PLATFORM_UNIX
-        static public TimeZoneInfo FindSystemTimeZoneById(string id) {
-            // UNIXTODO
-            throw new NotImplementedException();
+        static public TimeZoneInfo FindSystemTimeZoneById(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException("id");
+            }
+
+            // Special case for Utc since there is code that expects only 1 instance of the UTC time zone info to be created.
+            // see CachedData.GetCorrespondingKind()
+            if (String.Compare(id, c_utcId, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return TimeZoneInfo.Utc;
+            }
+
+            // UNIXTODO: use CachedData
+
+            string timeFilePath = Path.Combine(c_timeZoneDirectory, id);
+            if (!File.Exists(timeFilePath))
+            {
+                throw new TimeZoneNotFoundException(Environment.GetResourceString("TimeZoneNotFound_MissingRegistryData", id));
+            }
+
+            byte[] rawData = File.ReadAllBytes(timeFilePath);
```

Comment:
While it'd be very rare, the file could theoretically be deleted between the Exists check about and our attempt to read all of its data.  Plus there could be other issues that cause an exception to be thrown, such as if we don't have access to the file for some reason.  If the goal in such cases is to throw a TimeZoneNotFoundException, seems like this call should be wrapped in a try/catch that does the appropriate translation.